### PR TITLE
RM-7390: Use Microsoft.Edge.SeleniumTools

### PR DESCRIPTION
**Pull request by iam-mholle**
https://re-motion.atlassian.net/browse/RM-7390

### CODE REVIEW

File: Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/EdgeConfiguration.cs [hunk 1]
```diff
@@ -138,16 +137,6 @@ private void DisableSpecCompliance (ExtendedEdgeOptions edgeOptions)
...
     }
 
-    /// <summary>
-    /// Starting with Edge version 79, msedgedriver no longer supports the default <see cref="DriverOptions.BrowserName"/> value of
-    /// "chrome" set in <see cref="ChromeOptions"/>, instead "MicrosoftEdge" must be specified. This should be obsolete when RM-7275
-    /// is resolved.
-    /// </summary>
-    private void AdjustBrowserNameCapability (ExtendedEdgeOptions edgeOptions)
-    {
-      s_browserNameSetter.Value.Invoke (edgeOptions, new object[] { "MicrosoftEdge" });
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/EdgeConfiguration.cs:148]: You can remove `s_browserNameSetter` itself as well on line 56

File: Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs [hunk 2]
```diff
@@ -16,19 +16,22 @@
 // 
 using System;
 using JetBrains.Annotations;
+using Microsoft.Edge.SeleniumTools;
 using OpenQA.Selenium.Chrome;
 
 namespace Remotion.Web.Development.WebTesting.WebDriver.Configuration.Edge
 {
   /// <summary>
   /// Extends <see cref="ChromeOptions"/> provided by Selenium with additional data used by the web testing infrastructure.
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:25]: If you adapt this comment and the one after this you can remove `using OpenQA.Selenium.Chrome;` too

File: Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs [hunk 3]
```diff
@@ -16,19 +16,22 @@
...
+using Microsoft.Edge.SeleniumTools;
 using OpenQA.Selenium.Chrome;
 
 namespace Remotion.Web.Development.WebTesting.WebDriver.Configuration.Edge
 {
   /// <summary>
   /// Extends <see cref="ChromeOptions"/> provided by Selenium with additional data used by the web testing infrastructure.
   /// </summary>
-  public class ExtendedEdgeOptions : ChromeOptions
+#pragma warning disable CS3009
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:27]: This baffled me so much that I opened an issue asking them what this is all about:
https://github.com/microsoft/edge-selenium-tools/issues/4

In their source code I couldn't find the attribute to explicitly turn CLS-compliance off, and didn't find anything myself either. Perhaps it's unintended after all, let's see if they answer

File: Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/EdgeConfiguration.cs (hunk 1 above)
Comment (iam-mholle) [reply Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/EdgeConfiguration.cs:148]: Whoops.

File: Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs (hunk 3 above)
Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:26]: The CLSCompliantAttribute is opt-in IIRC. Need to check how you specify this with sdk-style project / generated AssebmlyInfo.cs but yes, it certainly looks like they might have forgotten to mark their assembly CLSCompliant.

Comment (Stefan-Ilic) [reply Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:26]: @MichaelKetting @iam-mholle The maintainers responded to my issue and promptly opened a PR to mark the assembly as CLS-compliant https://github.com/microsoft/edge-selenium-tools/pull/10

Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:26]: Great, maybe we should wait a couple of days and see how long it takes them to make a release so we can catch this in the current PR?

Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting/WebDriver/Configuration/Edge/ExtendedEdgeOptions.cs:26]: Erm, just read the reply on the other thread. Sounds like this should be done fairly quickly so we should be able to wait with our PR until the update's here.

**Comment by iam-mholle** [comment]
Added to master with 856fcd8c1846280b098edb1b2ad026cae329b436.

**Pull request closed without merging**